vpr/src/place/place_constraints.cpp

```diff
@@ -82,37 +82,95 @@ PartitionRegion constrained_macro_locs(const t_pl_macro& pl_macro) {
 
                 t_pl_loc min_pl_loc(reg_rect.xmin(), reg_rect.ymin(), block_regions[i].get_sub_tile());
 
-                t_pl_loc modified_min_pl_loc = min_pl_loc + offset;
+                t_pl_loc modified_min_pl_loc = min_pl_loc - offset;
 
                 t_pl_loc max_pl_loc(reg_rect.xmax(), reg_rect.ymax(), block_regions[i].get_sub_tile());
 
-                t_pl_loc modified_max_pl_loc = max_pl_loc + offset;
+                t_pl_loc modified_max_pl_loc = max_pl_loc - offset;
 
                 modified_reg.set_region_rect(modified_min_pl_loc.x, modified_min_pl_loc.y, modified_max_pl_loc.x, modified_max_pl_loc.y);
                 //check that subtile is not an invalid value before changing, otherwise it just stays -1
-                if (block_regions[i].get_sub_tile() != -1) {
+                if (block_regions[i].get_sub_tile() != NO_SUBTILE) {
                     modified_reg.set_sub_tile(modified_min_pl_loc.sub_tile);
                 }
 
                 modified_pr.add_to_part_region(modified_reg);
             }
 
             if (num_constrained_members == 1) {
-                macro_pr = modified_pr;
+                macro_head_pr = modified_pr;
             } else {
-                macro_pr = intersection(macro_pr, modified_pr);
+                macro_head_pr = intersection(macro_head_pr, modified_pr);
             }
         }
     }
 
     //if the intersection is empty, no way to place macro members together, give an error
-    if (macro_pr.empty()) {
+    if (macro_head_pr.empty()) {
         VPR_ERROR(VPR_ERROR_PLACE, " \n Feasible floorplanning constraints could not be calculated for the placement macro.\n");
     }
 
+    return macro_head_pr;
+}
+
+PartitionRegion update_macro_member_pr(PartitionRegion& head_pr, const t_pl_offset& offset) {
```

Comment:
Good for a function to comment to note that the tightest possible constraint (implied by the whole chain) must already have been set on the head of the macro that you pass in.